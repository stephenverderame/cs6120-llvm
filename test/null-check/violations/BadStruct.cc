// NOLINTBEGIN

#include <cassert>
#include <cstdint>
#include <cstdio>

struct Bar {
    int x;
    char y;
    int64_t data[20];
};

int main()
{
    Bar b;
    b.x = 10;
    b.y = 20;
    for (int i = 0; i < 25; ++i) {
        b.data[i] = i;
    }
}
// NOLINTEND