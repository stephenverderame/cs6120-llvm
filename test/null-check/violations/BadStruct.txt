[31mSafety Violation[00m: Use of potentially null pointer in store at BadStruct.cc:19
Because an index could not be proven to be in range, the following counterexample was found:
	[1mi[0m = 22
