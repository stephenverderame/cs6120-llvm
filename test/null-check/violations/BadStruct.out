[31mSafety Violation[00m: Use of potentially null pointer in store at BadStruct.cc:19
